src/vm/types.rs

```diff
@@ -912,7 +939,11 @@ impl TypeSignature {
         if let SymbolicExpressionType::AtomValue(Value::Int(max_len)) = &type_args[0].expr {            
             let atomic_type_arg = &type_args[type_args.len()-1];
             let atomic_type = TypeSignature::parse_type_repr(atomic_type_arg, false)?;
-            TypeSignature::new_list(atomic_type.atomic_type, *max_len, dimension)
+            if let TypeSignature::Atom(atomic_type) = atomic_type {
+                TypeSignature::new_list(atomic_type, *max_len, dimension)
+            } else {
+                panic!("Parser should not have returned a non-atom")
```

Comment:
I take it that somewhere earlier, the VM made sure that the value of `X` in `(list X Y)` or `(list X Y Z)` was an atom?

Reply:
Yeah -- `TypeSignature::parse_type_repr` takes a boolean argument indicating whether or not it's okay to return a list type, and will throw an InvalidTypeDescription error if it does.

Reply:
Gotcha.